policy-controller/k8s/index/src/inbound/routes.rs

```diff
@@ -0,0 +1,169 @@
+use anyhow::Result;
+use linkerd_policy_controller_core::POLICY_CONTROLLER_NAME;
+use linkerd_policy_controller_k8s_api::{gateway, policy};
+use std::fmt;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct RouteBinding<R> {
+    pub parents: Vec<ParentRef>,
+    pub route: R,
+    pub statuses: Vec<Status>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ParentRef {
+    Server(String),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Status {
+    pub parent: ParentRef,
+    pub conditions: Vec<Condition>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Condition {
+    pub type_: ConditionType,
+    pub status: bool,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ConditionType {
+    Accepted,
+}
+
+#[derive(Clone, Debug, thiserror::Error)]
+pub enum InvalidParentRef {
+    #[error("HTTPRoute resource may not reference a parent Server in an other namespace")]
+    ServerInAnotherNamespace,
```

Comment:
quick thought on error handling, should we update the enum variants here to encode the route type in the error message instead of relying on a hardcoded `HTTPRoute`?

My reasoning: if `RouteBinding` is generic over the type of route it composes with, then we may have a `ServerInAnotherNamespace` error displayed for a gRPC route, and it might incorrectly point that it's an HTTPRoute instead? We could probably also do this in the follow-up where gRPC routes are wired up in the inbound index 🤔

Reply:
Updated error message to be less specific about the resource kind.  We log the resource kind at the point where we emit this log, so the error message doesn't need to contain it.